### DISCUSSION TITLE
Current User.OAuthToken  always empty

### DISCUSSION THREAD

**Discussion by alphaadidas** [Budibase/budibase#9158] [Help]
Using Auth0 setup, the "{{ Current User.OAuthToken }}" is always empty.

Checked the DB directly, and the full auth0 blob is present.

env: self hosted.   v2.2.9

Is this an known issue? or just expected for the opensource version?

**Comment by melohagan** [comment]
Hey @alphaadidas 

You checked the DB? `{{ Current User.OAuthToken }}` is only populated when a user logs in with OAuth: https://docs.budibase.com/docs/authentication-and-sso

**Reply by alphaadidas** [reply]
Yes.. as i said in the original post "Checked the db directly, and the full auth0 blob is present" .. :(

**Reply by melohagan** [reply]
Sorry @alphaadidas I meant that I didn't understand what you meant by checking the DB - which DB?

If you mean the Users table, I don't believe that token is stored there - it may just be stored as a local session cookie of some sort.

I'm probably not the best person to answer this question though; I'll ask around.

**Reply by alphaadidas** [reply]
Im using all the provided documentation, and nothing is working.  OAuthToken concept/idea/approach is un-usable :(   ... i have to create a Machine2Machine flow , and hack a way to create a 'personate' token.

**Reply by melohagan** [reply]
Hey @alphaadidas

We stepped through an Auth0 setup and found no issues. 

In the REST Connector we set the authentication like so:
![Screenshot 2023-02-14 at 13 18 04](https://user-images.githubusercontent.com/101575380/218750161-a8b3b10b-6fed-4fdd-ad63-13c0974b5872.png)

![Screenshot 2023-02-14 at 13 18 38](https://user-images.githubusercontent.com/101575380/218750279-c1e595fa-e628-45c2-a992-382551184f6d.png)

![Screenshot 2023-02-14 at 13 18 55](https://user-images.githubusercontent.com/101575380/218750353-35d208c9-fab7-4281-98d9-7bccb988b36f.png)
**Make sure to click Save**

![Screenshot 2023-02-14 at 13 19 30](https://user-images.githubusercontent.com/101575380/218750505-405bfe5f-03d2-45cc-b8f5-7431c1163aac.png)
**Set the auth in the connector query**

**Reply by alphaadidas** [reply]
Yep.. did ALL of that.  :( :( :(

**Reply by melohagan** [reply]
Hmm, it might help if you were able to record a video of your setup, and the steps you're taking.

**Reply by alphaadidas** [reply]
Let me recreate the whole flow, and see if i missed a step.

**Reply by alphaadidas** [reply]
Still nothing... same steps. :(

**Comment by alphaadidas** [comment]
I ended up re-creating the app from scratch.. then the SSO token started to show up.    So not sure what the issue is, with the orignal app i was building.

**Comment by bemble** [comment]
Hi, I'm also experiencing the same issue. I tried to create a new app from scratch, but it still empty.
Even the value helper shows an empty value:
<img width="385" alt="Capture d’écran 2025-05-23 à 18 04 15" src="https://github.com/user-attachments/assets/a480e9f5-0edd-4d1b-b4aa-235261e6e9d4" />
Any idea?
